codalab/worker/bundle_service_client.py

```diff
@@ -43,6 +43,10 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def parse_dynamic_error_type(error):
+    return error if isinstance(error, str) else error.decode()
```

Comment:
Looks good. I remember I had initially had this in the python 2->3 migration but removed it because I thought it was unnecessary (and wanted to avoid the str-bytes multiple types case) -- but it appears that in this case, we have to do this for HTTPError

Reply:
Yes, I think this is mainly due to we manually throw an HTTPError in our code base. I tried to look for methods to sending bytes instead through `urllib.error.HTTPError`, but haven't find anything useful yet.